source/code/plugins/collectd_lib.rb

```diff
@@ -38,21 +38,27 @@ def transform(record, hostname)
 
     #add additional meta such as ObjectName, InstanceName, etc.
     def transform_and_wrap(record, hostname)
-       data_items = transform(record, hostname)
-       if (data_items != nil and data_items.size > 0)
-         wrapper = {
-          "DataType"=>"LINUX_PERF_BLOB",
-          "IPName"=>"LogManagement",
-          "DataItems"=>data_items
-        }
-         return wrapper
-       else
-         # no data items, send a empty array that tells ODS
-         # output plugin to not the data
-         return {}
-       end
+      if record !=nil and record["error_class"]=="Fluent::BufferQueueLimitError"
```

Comment:
The error_class I used is part of the collectd record.

016-05-24 15:12:36 -0700 [warn]: dump an error event: error_class=NoMethodError error="undefined method `each_index' for nil:NilClass" tag="fluent.warn" time=1464127956 record={"error_class"=>"Fluent::BufferQueueLimitError", "error"=>"queue size exceeds limit", "tag"=>"oms.collectd", "message"=>"emit transaction failed: error_class=Fluent::BufferQueueLimitError error=\"queue size exceeds limit\" tag=\"oms.collectd\"", "Timestamp"=>"2016-05-24T22:12:36.278Z"}